components/Home/DevIncentives/requests.jsx

```diff
@@ -96,27 +108,44 @@ const getEpochLength = async () => {
   return parseInt(response, 10);
 };
 
-export const canShowCheckpoint = async () => {
+export const canShowCheckpoint = async (
+  checkOnlyIfEpochLenIsGreater = false,
+) => {
   try {
     const epCounter = await getEpochCounter();
-    const epTokenomics = await getEpochTokenomics({
-      lastPoint: Number(epCounter) - 1,
-    });
+    const epTokenomics = await getEpochTokenomics(Number(epCounter) - 1);
     const epochLen = await getEpochLength();
     const blockTimestamp = await getBlockTimestamp();
-    const { endTime } = epTokenomics;
+    const timeDiff = blockTimestamp - epTokenomics.endTime;
 
-    if (blockTimestamp - endTime >= epochLen) {
-      return true;
-    }
-    return false;
+    return checkOnlyIfEpochLenIsGreater
+      ? timeDiff > epochLen
+      : timeDiff >= epochLen;
   } catch (error) {
     console.error(error);
   }
 
   return false;
 };
 
+export const getActualEpochTimeLength = async () => {
+  try {
+    const epCounter = await getEpochCounter();
+    const epTokenomics = await getEpochTokenomics(Number(epCounter) - 1);
+    const epochLen = await getEpochLength();
+    const blockTimestamp = await getBlockTimestamp();
+    const timeDiff = blockTimestamp - epTokenomics.endTime;
+
+    return timeDiff > epochLen
+      ? timeDiff
+      : epochLen;
+  } catch (error) {
+    console.error(error);
+  }
+
+  return 0;
+};
```

Comment:
@mohandast52 please combine or optimize the logic. For this file we need the second one - the actual time in seconds, that was the issue for miscalculation.